wgpu-hal/src/metal/device.rs

```diff
@@ -630,9 +630,14 @@ impl crate::Device<super::Api> for super::Device {
     unsafe fn create_shader_module(
         &self,
         _desc: &crate::ShaderModuleDescriptor,
-        shader: crate::NagaShader,
+        shader: crate::ShaderInput,
     ) -> Result<super::ShaderModule, crate::ShaderError> {
-        Ok(super::ShaderModule { raw: shader })
+        match shader {
+            crate::ShaderInput::NagaShader(raw) => Ok(super::ShaderModule { raw }),
+            crate::ShaderInput::SpirVShader(_) => Err(crate::ShaderError::Compilation(
```

Comment:
I think we should just straight panic here. The general approach is to return errors only where the user can't anticipate a failure. In case of SPIR-V, they very well know if they passed the feature to device request or not. So a failure here is not a surprise.

Reply:
That's true, but "panic on missing feature" isn't something I've seen around in WGPU. That said, valid.

Reply:
Actually, that's a prime target for unreachable!(). We shouldn't be here - we only get here if Metal device somehow reported it supports SPIRV_SHADER_MODULES.

Reply:
Right. wgpu doesn't panic, but also - it validates this. wgpu-hal can panic just fine if reaching something that shouldn't be reachable.